cartographer/mapping/internal/3d/pose_graph_3d.cc

```diff
@@ -952,6 +952,7 @@ void PoseGraph3D::SetLandmarkPose(const std::string& landmark_id,
   AddWorkItem([=]() LOCKS_EXCLUDED(mutex_) {
     absl::MutexLock locker(&mutex_);
     data_.landmark_nodes[landmark_id].global_landmark_pose = global_pose;
+    data_.landmark_nodes[landmark_id].frozen = true;
```

Comment:
This is an interesting moment. What if someone uses `SetLandmarkPose` to specify  starting point for optimization? Should it be always frozen then? Same in 2d case.

Reply:
So far I did not know that somebody would use it like this. But we could easily deal with it by adding a `bool frozen=false` to the function arguments.

I think whenever they are read from a .pbstream file they should be frozen.

We could also think about what should happen to the landmarks when  `freezeTrajectory(..)` is called. Personally, since landmarks are not directly associated to a trajectory, I think this call should not affect the landmarks.

If somebody ever needs to freeze individual landmarks a method `freezeLandmark(..)` should be implemented.

Reply:
Sometimes we load a pbstream to continue optimizing jointly everything we have. Then it should not be frozen.  Please, add `bool frozen=false` to the function arguments.

I agree with `FreezeLandmark(...)`.

Reply:
okay, so should I in this case also add an option `--freeze_landmarks` similar to `--freeze_trajectory`? Or should I keep the setting in `map_builder.cc`` to `frozen = true`, which finally corresponds to the original state?

Reply:
for now i would leave the flags as they are. At least for this PR.

Reply:
I made the change to the function signature. I'm setting `frozen = true` in `map_builder.cc` but otherwise the function arguments defaults to false.

I did not fully look into the _cloud_ methods, they still work as expected, but I don't understand it well enough to reason how frozen should be set there or how it should be added to the protobuf requests.

If this is necessary, does somebody else have time to do it?